bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java

```diff
@@ -3729,7 +3729,8 @@ boolean hitTestSelection (long hItem, int x, int y) {
 	int state = (int)OS.SendMessage (handle, OS.TVM_GETITEMSTATE, hItem, OS.TVIS_SELECTED);
 	int detail = (state & OS.TVIS_SELECTED) != 0 ? SWT.SELECTED : 0;
 	Event event = sendMeasureItemEvent (item, order [index [0]], hDC, detail);
-	if (event.getBoundsInPixels ().contains (x, y)) result = true;
+	int zoom = getZoom();
+	if (event.getBounds ().contains (DPIUtil.scaleDown(x, zoom), DPIUtil.scaleDown(y, zoom))) result = true;
```

Comment:
I am not sure how relevant that is, but couldn't this slightly change behavior because of different rounding? In general, I would expect data loss when scaling up the bounds instead of scaling down the position to check for being within the bounds.

Reply:
There is a second place, where this is done. I don't expect it to have a real measurable effect by the user, but I will adapt it, so the bounds a upscaled to don't change the logic at all.

Reply:
I agree that there will probably not be a (noticable) effect, but 
1. we never know ;-)
2. we should keep refactoring separate from (potential) functional changes.